### PR TITLE
Add new Coffee Shop Kata and Coffee Shop Kata solution module

### PR DESCRIPTION
**Pull request by aqsa505**
Signed-off-by: Aqsa Malik [aqsa.malik@bnymellon.com](mailto:aqsa.malik@bnymellon.com)

### CODE REVIEW

File: coffee-shop-kata/old-java-features/src/test/java/CoffeeShopTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    @Test
+    public void testDonutGetters() {
+        assertEquals(donut1.getDonutType(), GLAZED);
+    }
+
+    @Test
+    public void testSealedClasses(){
+        assertFalse(CoffeeDrink.class.isSealed());
```
Comment (tedyoung) [inline coffee-shop-kata/old-java-features/src/test/java/CoffeeShopTest.java:115]: isSealed() isn't available in Java 8, only 17+.

File: coffee-shop-kata/old-java-features/src/test/java/CoffeeShopTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    @Test
+    public void getDrinkItems() {
+        List<String> expected = List.of("HOT Americano", "HOT CARAMEL Latte with ALMOND_MILK",
```
Comment (tedyoung) [inline coffee-shop-kata/old-java-features/src/test/java/CoffeeShopTest.java:120]: List.of is Java 9+, not available in Java 8.

File: coffee-shop-kata/old-java-features/src/test/java/CoffeeShopTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import static bnymellon.codekatas.coffeekata.food.BagelType.EVERYTHING;
+import static bnymellon.codekatas.coffeekata.food.CookieType.CHOCOLATE_CHIP;
+import static bnymellon.codekatas.coffeekata.food.DonutType.GLAZED;
+import static bnymellon.codekatas.coffeekata.food.SpreadType.HERB_GARLIC_CREAM_CHEESE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
```
Comment (tedyoung) [inline coffee-shop-kata/old-java-features/src/test/java/CoffeeShopTest.java:34]: Missing import for assertFalse.

File: coffee-shop-kata/old-java-features/src/test/java/CoffeeShopTest.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import bnymellon.codekatas.coffeekata.CoffeeShopOrder;
+import bnymellon.codekatas.coffeekata.Item;
+import bnymellon.codekatas.coffeekata.food.Bagel;
```
Comment (tedyoung) [inline coffee-shop-kata/old-java-features/src/test/java/CoffeeShopTest.java:19]: Missing import for bnymellon.codekatas.coffeekata.beverage.CoffeeDrink

**Review by tedyoung — COMMENTED** [review]
Hope it's OK to comment on this as I wanted to try it out.

File: coffee-shop-kata/new-java-features/src/main/java/bnymellon/codekatas/coffeekata/beverage/DrinkTemperature.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bnymellon.codekatas.coffeekata.beverage;
+
+public enum DrinkTemperature
+{HOT, ICED}
```
Comment (donraab) [inline coffee-shop-kata/new-java-features/src/main/java/bnymellon/codekatas/coffeekata/beverage/DrinkTemperature.java:20]: Inconsistent formatting.

File: coffee-shop-kata/new-java-features/src/main/java/bnymellon/codekatas/coffeekata/beverage/FlavorSyrup.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+ * limitations under the License.
+ */
+
+package bnymellon.codekatas.coffeekata.beverage;
+
+public enum FlavorSyrup
+{
+    VANILLA,
+    CHOCOLATE,
+    CARAMEL}
```
Comment (donraab) [inline coffee-shop-kata/new-java-features/src/main/java/bnymellon/codekatas/coffeekata/beverage/FlavorSyrup.java:23]: Inconsistent formatting.

File: coffee-shop-kata/old-java-features/pom.xml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
```
Comment (donraab) [inline coffee-shop-kata/old-java-features/pom.xml:52]: While you are using old Java features, I don't think you need to keep source and target configuration at Java 8.

File: coffee-shop-kata/old-java-features/src/main/java/bnymellon/codekatas/coffeekata/CoffeeShopOrder.java [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    public List<String> getFoodItemsForOrder()
+    {
+        List<String> foodItems = new ArrayList<>();
+        for (Item item : this.orderItems)
+        {
+            if (item instanceof Bagel)
+            {
+                Bagel bagel = (Bagel) item;
+                foodItems.add(bagel.getBagelType() + " bagel with " + bagel.getSpreadType());
+            } else if (item instanceof Cookie)
```
Comment (donraab) [inline coffee-shop-kata/old-java-features/src/main/java/bnymellon/codekatas/coffeekata/CoffeeShopOrder.java:68]: Inconsistent formatting on else if

File: coffee-shop-kata/old-java-features/src/main/java/bnymellon/codekatas/coffeekata/food/Bagel.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        return bagelType;
+    }
+
+    public SpreadType getSpreadType()
+    {
+        return spreadType;
+    }
+
+    @Override
+    public boolean equals(Object obj)
```
Comment (donraab) [inline coffee-shop-kata/old-java-features/src/main/java/bnymellon/codekatas/coffeekata/food/Bagel.java:54]: You override equals but not hashCode.

File: coffee-shop-kata/old-java-features/src/main/java/bnymellon/codekatas/coffeekata/food/Cookie.java [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        return warmed;
+    }
+
+    public CookieType getCookieType()
+    {
+        return cookieType;
+    }
+
+    @Override
+    public boolean equals(Object obj)
```
Comment (donraab) [inline coffee-shop-kata/old-java-features/src/main/java/bnymellon/codekatas/coffeekata/food/Cookie.java:47]: You override equals but not hashCode

File: coffee-shop-kata/old-java-features/src/main/java/bnymellon/codekatas/coffeekata/food/Donut.java [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+        return 1.75;
+    }
+
+    public DonutType getDonutType()
+    {
+        return this.donutType;
+    }
+
+    @Override
+    public boolean equals(Object obj)
```
Comment (donraab) [inline coffee-shop-kata/old-java-features/src/main/java/bnymellon/codekatas/coffeekata/food/Donut.java:40]: You override equals but not hashCode

File: coffee-shop-kata/pom.xml [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+    <groupId>bnymellon.codekatas</groupId>
+    <artifactId>coffee-shop-kata</artifactId>
+    <version>1.0.0</version>
+
+    <name>coffee-shop-kata</name>
+    <packaging>pom</packaging>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>17</maven.compiler.source>
```
Comment (donraab) [inline coffee-shop-kata/pom.xml:17]: Should this be set to 20 instead of 17?

**Review by donraab — CHANGES_REQUESTED** [review]
Hi @aqsa505, looks mostly good. I have some changes request inline. Thanks!

**Review by donraab — APPROVED** [review]
LGTM

**Pull request merged**